방 리스트 조회 필터링 조건 추가 및 현재 방 인원 수 조회되도록 변경

hyoguoo opened the pull request:
## 📝작업 내용

> 이번 PR에서 작업한 내용을 간략히 설명해주세요

- [x] 방 리스트 조회 필터 조건 추가
- [x] 방 - 멤버 OneToMany 관계 추가
- [x] 방 리스트 조회 동적 쿼리 작성

## 💬리뷰 시 요구사항 (선택)

> 리뷰어가 특별히 봐주었으면 하는 부분이 있다면 작성해주세요

hyoguoo committed:
feat #83: 방 리스트 조회 요청 정보 검색어/상태/타입/언어 추가

hyoguoo committed:
feat #83: 방 리스트 조회 querydsl 작성

hyoguoo committed:
chore #83: static 삭제

hyoguoo committed:
fix #83: Solve History 리스트 조회 시 delete not null만 조회하도록 수정

hyoguoo committed:
chore #83: 코드 리포멧

hyoguoo committed:
mod #83: 동적쿼리 적용을 위한 조회 메서드 변경 및 PageResponse 객체 적용

hyoguoo committed:
feat #83: 현재 방 인원 수를 가져오기 위한 OneToMany 관계 추가

hyoguoo committed:
feat #83: 현재 방 인원 수도 포함해서 반환

GiHoo commented on src/main/java/ei/algobaroapi/domain/room/controller/RoomControllerDocImpl.java:
```diff
@@ -61,7 +62,9 @@ public RoomDetailResponseDto getRoomByShortUuid(
     @Override
     @PostMapping("rooms/codes/{roomShortUuid}")
-    public RoomDetailResponseDto startCodingTest(@PathVariable(name = "roomShortUuid") String roomShortUuid) {
+    public RoomDetailResponseDto startCodingTest(
+            @PathVariable(name = "roomShortUuid") String roomShortUuid
+    ) {
```
👍

hyoguoo committed:
fix #83: 언어 필터 조건 미요청 시 오류 수정

GiHoo commented on src/main/java/ei/algobaroapi/domain/room/domain/RoomRepository.java:
```diff
@@ -6,7 +6,7 @@
+public interface RoomRepository extends JpaRepository<Room, Long>, RoomRepositoryQuery {
 
     @Query("SELECT r FROM Room r WHERE r.roomUuid LIKE CONCAT(:roomShortUuid, '%')")
     Optional<Room> findByRoomUuidStartingWith(String roomShortUuid); // TODO: 방 참여 기능 구현 후 쿼리 수정
```
리뷰하다가 알게된 TODO..

GiHoo commented on src/main/java/ei/algobaroapi/domain/room/domain/RoomRepositoryQueryImpl.java:
👀

hyoguoo committed:
fix #83: Page 조건 미요청 시 default 값 설정

GiHoo commented on src/main/java/ei/algobaroapi/domain/room/dto/request/RoomListRequestDto.java:
```diff
@@ -14,4 +17,16 @@ public class RoomListRequestDto {
+    @Schema(description = "방 제목 검색어", example = "초보방")
+    private String searchTitle;
+
+    @Schema(description = "방 상태(입장 가능으로 사용 시 RECRUITING, 체크 해제 시 요청 x)", example = "RECRUITING")
+    private RoomStatus roomStatus;
```
오.. 전 진행 중인 방에 대해서는 안보여준다고 생각했는데 
이것에 대한 방 입장 로직도 있어야하겠군요

hyoguoo replied:
혹시 몰라서 해당 방식으로 진행 + 로직 처리가 편해서 저 방식으로 했습니다ㅎㅎ

GiHoo commented on src/main/java/ei/algobaroapi/domain/room/service/RoomServiceImpl.java:
```diff
@@ -32,13 +33,17 @@ public class RoomServiceImpl implements RoomService {
+                roomRepository.findListPage(
+                        request,
+                        PageRequest.of(
+                                Optional.ofNullable(request.getPage()).orElse(0),
+                                Optional.ofNullable(request.getSize()).orElse(6)
```
기본 값 좋네요 👍

hyoguoo replied:
Request DTO에서 해결하려 했으나 오류가 발생하여 추후에 수정하고자 합니다..

GiHoo reviewed (approved):
동적 쿼리 작성해주시느라 고생하셨습니다!
👍  👍

hyoguoo committed:
fix #83: fetch join 시 많은 데이터를 불러올 수 있는 위험 수정

HHH90003004: firstResult/maxResults specified with collection fetch; applying in memory

위 에러 발생하여 fetch join 제거 후

default_batch_fetch_size 값 주어 해결
